projects/dashboards/ebv-analyzer/src/app/data-selection.service.ts

```diff
@@ -120,3 +121,32 @@ export class DataSelectionService {
         );
     }
 }
+
+/**
+ * Checks if the `timeSteps` contain information about months or days.
+ * For instance, for the steps 2019-01-01, 2019-02-01, 2019-03-01, the format would be 'YYYY-MM'.
+ */
+export function estimateTimeFormat(timeSteps: Array<Time>): string {
+    let useMonth = false;
+    let useDay = false;
+    for (const timeStep of timeSteps) {
+        // in range [1, 31]
+        if (timeStep.start.date() > 1) {
+            useDay = useMonth = true;
+            break;
+        }
+        // in range [0, 11]
+        if (timeStep.start.month() > 0) {
+            useMonth = true;
```

Comment:
what if we have a yearly dataset that doesn't start in January?
e.g. [1.6.2000, 1.6.2001, ...]

do we ignore this for now?

Reply:
Since June seems to be a relevant information, I think so?